x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheDirectory.java

```diff
@@ -383,4 +396,17 @@ private static boolean assertFileChannelOpen(FileChannel fileChannel) {
         assert fileChannel.isOpen();
         return true;
     }
+
+    private static String getExtension(String name) {
+        int i = name.lastIndexOf('.');
+        if (i == -1) {
+            return "";
+        }
+        return name.substring(i + 1);
+    }
+
+    private boolean isBlackListedFromCache(String name) {
+        String ext = getExtension(name);
```

Comment:
Let's just use` IndexFileNames.getExtension()` and in case of no extension always cache if `name.startsWith(IndexFileNames.SEGMENTS) || name.equals(IndexFileNames.OLD_SEGMENTS_GEN)` ?

Reply:
fixed in a2c5489